sdks/python/apache_beam/io/tfrecordio_test.py

```diff
@@ -169,157 +160,237 @@ def _write_lines(self, sink, path, lines):
+    with TempDir() as temp_dir:
+      path = temp_dir.create_temp_file('result')
+      _write_file_gzip(path, FOO_BAR_RECORD_BASE64)
+      with TestPipeline() as p:
+        result = (p
+                  | ReadFromTFRecord(
+                      path, compression_type=CompressionTypes.GZIP))
+        assert_that(result, equal_to(['foo', 'bar']))
 
-class TestReadFromTFRecordSource(TestTFRecordSource):
+  def test_process_gzip_auto(self):
+    with TempDir() as temp_dir:
+      path = temp_dir.create_temp_file('result.gz')
+      _write_file_gzip(path, FOO_BAR_RECORD_BASE64)
+      with TestPipeline() as p:
+        result = (p
+                  | ReadFromTFRecord(
+                      path, compression_type=CompressionTypes.AUTO))
+        assert_that(result, equal_to(['foo', 'bar']))
+
+
+class TestReadAllFromTFRecord(TestCase):
+
+  def _write_glob(self, temp_dir, prefix):
+    path1 = temp_dir.create_temp_file(prefix)
+    path2 = temp_dir.create_temp_file(prefix)
+    path3 = temp_dir.create_temp_file(prefix)
+    _write_file(path1, FOO_BAR_RECORD_BASE64)
+    _write_file(path2, FOO_BAR_RECORD_BASE64)
+    _write_file(path3, FOO_BAR_RECORD_BASE64)
+
+  def test_process_single(self):
+    with TempDir() as temp_dir:
+      path = temp_dir.create_temp_file('result')
+      _write_file(path, FOO_RECORD_BASE64)
+      with TestPipeline() as p:
+        result = (p
+                  | Create([path])
+                  | ReadAllFromTFRecord(
+                      coder=coders.BytesCoder(),
+                      compression_type=CompressionTypes.AUTO))
+        assert_that(result, equal_to(['foo']))
+
+  def test_process_multiple(self):
+    with TempDir() as temp_dir:
+      path = temp_dir.create_temp_file('result')
+      _write_file(path, FOO_BAR_RECORD_BASE64)
+      with TestPipeline() as p:
+        result = (p
+                  | Create([path])
+                  | ReadAllFromTFRecord(
+                      coder=coders.BytesCoder(),
+                      compression_type=CompressionTypes.AUTO))
+        assert_that(result, equal_to(['foo', 'bar']))
+
+  def test_process_glob(self):
+    with TempDir() as temp_dir:
+      self._write_glob(temp_dir, 'result')
+      glob = temp_dir.get_path() + os.path.sep + '*result'
+      with TestPipeline() as p:
+        result = (p
+                  | Create([glob])
+                  | ReadAllFromTFRecord(
+                      coder=coders.BytesCoder(),
+                      compression_type=CompressionTypes.AUTO))
+        assert_that(result, equal_to(['foo', 'bar'] * 3))
+
+  def test_process_multiple_globs(self):
+    with TempDir() as temp_dir:
+      self._write_glob(temp_dir, 'result1')
+      glob1 = temp_dir.get_path() + os.path.sep + '*result1'
+      self._write_glob(temp_dir, 'result2')
+      glob2 = temp_dir.get_path() + os.path.sep + '*result2'
+      self._write_glob(temp_dir, 'result3')
+      glob3 = temp_dir.get_path() + os.path.sep + '*result3'
```

Comment:
<!--new_thread; commit:d4254d9fe817a78ccd0f7704c87bc89f903add8c; resolved:0-->
nit: this could be done in a loop (for i in xrange(3): ...)

Reply:
Done.